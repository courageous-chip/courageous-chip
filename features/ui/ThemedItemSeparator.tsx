import { useTheme } from "@react-navigation/native";
import React, { FC } from "react";
import { StyleSheet, View } from "react-native";

export const ThemedItemSeparator: FC = function () {
  const {
    colors: { border: backgroundColor },
  } = useTheme();

  return <View style={[styles.separator, { backgroundColor }]} />;
};

const styles = StyleSheet.create({
  separator: {
    height: 1,
    width: "100%",
  },
});
