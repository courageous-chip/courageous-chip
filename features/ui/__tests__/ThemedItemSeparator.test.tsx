import React from "react";
import renderer from "react-test-renderer";

import { ThemedItemSeparator } from "../ThemedItemSeparator";

describe("<EmptyView />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ThemedItemSeparator />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
