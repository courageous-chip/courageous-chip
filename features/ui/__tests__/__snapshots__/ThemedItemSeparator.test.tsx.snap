// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyView /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 1,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgb(224, 224, 224)",
      },
    ]
  }
/>
`;
